### PR TITLE
[build] Enable clang-specific warnings only when using clang

### LINKED ISSUE #8447: Just updated source tree, build fails.
[ 28%] Building C object confu-deps/NNPACK/CMakeFiles/nnpack.dir/src/softmax-output.c.o
[ 28%] Building C object confu-deps/NNPACK/CMakeFiles/nnpack.dir/src/x86_64-fma/softmax.c.o
[ 28%] Building C object confu-deps/NNPACK/CMakeFiles/nnpack.dir/src/convolution-output.c.o
[ 28%] Building C object confu-deps/NNPACK/CMakeFiles/nnpack.dir/src/relu-input-gradient.c.o
[ 28%] Linking C static library ../../lib/libnnpack.a
[ 28%] Built target nnpack
make: *** [all] Error 2
Failed to run 'bash tools/build_pytorch_libs.sh --with-cuda --with-nnpack --with-mkldnn nccl caffe2 nanopb libshm gloo THD c10d'
(pytorch) developer@theano:~/pytorch$

### PR DESCRIPTION
**Pull request by ssnl**
Should fix #8447 

@smessmer on the `c10/CMakeList.txt` change.

### CODE REVIEW

**Comment by goldsborough** [comment]
Yeah this should fix those warnings. Ultimately we want to use `target_compile_options` and stuff but that's for the future

**Review by ezyang — APPROVED** [review]

**Comment by vadimkantorov** [comment]
just built master, still getting plenty of :
```
cc1: warning: command line option ‘-fvisibility-inlines-hidden’ is valid for C++/ObjC++ but not for C
```
while building aten
